lib/mongoose_push/service/apns.ex

```diff
@@ -5,77 +5,64 @@ defmodule MongoosePush.Service.APNS do
+    Notification.new(device_id, Map.get(request, :mode, :prod))
+    |> Notification.add_content_available(1)
+    |> maybe(:add_apns_topic, request[:topic])
+    |> add_data(request[:data])
     |> maybe_mutable_content(request[:mutable_content])
-    |> Notification.put_priority(@priority_mapping[request[:priority]])
+    |> maybe(:add_apns_priority, @priority_mapping[request[:priority]])
   end
 
-  def prepare_notification(device_id, request) do
+  def prepare_notification(device_id, request, pool) do
     # Setup non-silent notification
     alert = request.alert
+    default_topic = State.get_default_topic(pool)
 
-    %{
-      "alert" => %{
-        "title" => alert.title,
-        "body" => alert.body
-      },
-      "badge" => alert[:badge],
-      "category" => alert[:click_action],
-      "sound" => alert[:sound]
-    }
-    |> Notification.new(device_id, request[:topic], request[:data])
+    Notification.new(device_id, Map.get(request, :mode, :prod))
+    |> maybe(:add_apns_topic, request[:topic] || default_topic)
+    |> add_data(request[:data])
     |> maybe_mutable_content(request[:mutable_content])
-    |> Notification.put_priority(@priority_mapping[request[:priority]])
+    |> maybe(:add_apns_priority, @priority_mapping[request[:priority]])
+    |> Notification.add_title(alert.title)
+    |> Notification.add_body(alert.body)
+    |> maybe(:add_badge, alert[:badge])
+    |> maybe(:add_category, alert[:click_action])
+    |> maybe(:add_sound, alert[:sound])
   end
 
   @spec push(Service.notification(), String.t(), atom(), Service.options()) ::
           :ok | {:error, term}
-  def push(notification, _device_id, worker, opts \\ []) do
-    case APNS.push(notification, Keyword.merge([name: worker], opts)) do
-      {:ok, _state} ->
+  def push(notification, _device_id, pool, opts \\ []) do
+    case APNS.push(pool, notification, is_sync: true) do
+      :ok ->
         :ok
 
-      {:error, reason, _state} ->
+      {:error, reason} ->
         {:error, reason}
     end
   end
 
-  @spec workers({atom, Keyword.t()} | nil) :: list(Supervisor.Spec.spec())
-  def workers(nil), do: []
-
-  def workers({pool_name, pool_config}) do
-    pool_size = pool_config[:pool_size]
-
-    pool_config =
-      pool_config
-      |> construct_apns_endpoint_options()
-      |> announce_subject()
-      |> maybe_setup_default_apns_topic()
-
-    Enum.map(1..pool_size, fn id ->
-      worker_name = Pools.worker_name(:apns, pool_name, id)
-      worker_config = Config.config(worker_name, pool_config)
+  @spec supervisor_entry([Application.pool_info()] | nil) :: {module(), term()}
+  def supervisor_entry(pool_configs) do
+    {MongoosePush.Service.APNS.Supervisor, pool_configs}
+  end
 
-      Supervisor.Spec.worker(
-        Pigeon.APNSWorker,
-        [worker_config],
-        id: worker_name
-      )
-    end)
+  @spec choose_pool(MongoosePush.service(), MongoosePush.mode()) :: atom() | nil
+  def choose_pool(service, mode) do
```

Comment:
I don't think we need `service` argument here, as this is already specific for APNS.